tools/rc/usResourceCompiler.cpp

```diff
@@ -0,0 +1,634 @@
+  if (US_OK != retCode) return retCode;
+  
+  if (!mz_zip_writer_add_file(pZip, pArchive_name, pSrc_filename, pComment,
+                              comment_size, level_and_flags))
+  {
+    return US_MZ_ERROR_ADD_FILE;
+  }
+  return us_zip_writer_add_dir_entries(pZip, pArchive_name, archived_dirs);
+}
+
+static int us_zip_writer_add_from_zip_reader(mz_zip_archive *pZip, mz_zip_archive *pSource_zip,
+                                             mz_uint file_index, us_archived_names* archived_names,
+                                             us_archived_names* archived_dirs,
+                                             char* archiveName, mz_uint archiveNameSize)
+{
+  int retCode = 0;
+  
+  mz_uint numBytes = mz_zip_reader_get_filename(pSource_zip, file_index, archiveName, archiveNameSize);
+  if (numBytes > 1 && archiveName[numBytes-2] != '/')
+  {
+    retCode = us_archived_names_append(archived_names, archiveName);
+    if (US_OK != retCode) return retCode;
+    
+    if (!mz_zip_writer_add_from_zip_reader(pZip, pSource_zip, file_index))
+    {
+      return US_MZ_ERROR_ADD_FILE;
+    }
+  }
+  return us_zip_writer_add_dir_entries(pZip, archiveName, archived_dirs);
+}
+
+struct Custom_Arg : public option::Arg
+{
+  static void printError(const std::string& msg1, const option::Option& opt, const std::string& msg2)
+  {
+    std::cerr << "ERROR: " << msg1 << opt.name << msg2 << std::endl;
+  }
+  
+  static option::ArgStatus NonEmpty(const option::Option& option, bool msg)
+  {
+    if (option.arg != 0 && option.arg[0] != 0)
+      return option::ARG_OK;
+    
+    if (msg) printError("Option '", option, "' requires a non-empty argument\n");
+    return option::ARG_ILLEGAL;
+  }
+  
+  static option::ArgStatus Numeric(const option::Option& option, bool msg)
+  {
+    char* endptr = 0;
+    if (option.arg != 0 && strtol(option.arg, &endptr, 10)){};
+    if (endptr != option.arg && *endptr == 0)
+      return option::ARG_OK;
+    
+    if (msg) printError("Option '", option, "' requires a numeric argument\n");
+    return option::ARG_ILLEGAL;
+  }
+};
+
+#define US_PROG_NAME "usResourceCompiler"
+
+enum  OptionIndex { UNKNOWN, HELP, VERBOSE, BUNDLENAME, COMPRESSIONLEVEL, OUTFILE, RESFILE, MERGEZIP, APPENDBINARY };
+const option::Descriptor usage[] =
+{
+  {UNKNOWN,      0, "" , ""    , Custom_Arg::None, "\nUSAGE: " US_PROG_NAME " [options]\n\n"
+    "Options:" },
+  {HELP,         0, "h" , "help",Custom_Arg::None, " --help, -h  \tPrint usage and exit." },
+  {VERBOSE,         0, "v" , "verbose",Custom_Arg::None, " --verbose, -v  \tRun in verbose mode." },
+  {BUNDLENAME,    0, "b", "bundle-name", Custom_Arg::NonEmpty, " --bundle-name, -b \tThe bundle name as specified in the US_BUNDLE_NAME compile definition."},
+  {COMPRESSIONLEVEL,  0, "c", "compression-level", Custom_Arg::Numeric, " --compression-level, -c  \tCompression level used for zip . Value range is 0 to 9. Default value is 6." },
```

Comment:
In the `zip .` sub-string there seems to be a non-ascii character (try to copy / paste it into a text editor.